### ISSUE TITLE
s3-request-presigner getSignedUrl - expireIn appears to not have any effect

### ISSUE THREAD

**Issue by noamavner-ma** [aws/aws-sdk-js-v3#5577]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

Getting Token Expire after 24 hours instead of 1 week.
Had this running with 1 week fine using SDK V2. The creds are in the machine.

Using this code: 

```const params = {
      Bucket: this.bucketName,
      Key: filePath,
    }
    const getObjectCommand = new GetObjectCommand(params)
    return s3RequestPresigner.getSignedUrl(
      this.client,
      getObjectCommand,
      {
        expiresIn: 604800 // seven days
      },
    )
```
### SDK version number

"@aws-sdk/s3-request-presigner": "^3.451.0"

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

20.3.1

### Reproduction Steps

Use this code.

### Observed Behavior

Getting token expire before the time

### Expected Behavior

Should get token 1 week

**Comment by Omrilu-ma** [comment]
Hi @noamavner-ma 
Did you check the following?

**TLDR;**

**The two ways of expirations both determine how long the URLs are valid. If the signature is valid for 15 minutes but the session has only 5 minutes remaining the effective duration is only 5 minutes. This is usually a problem for URLs signed close to the end of the session which is one of the worst kind of bugs to debug.**


**Expiration**
Another problem is the expiration. Signed URLs expire in two ways:

First, the signature can expire. When you sign a URL, you specify how long it will be valid:
```
const url = s3.getSignedUrl("getObject", {
	Expires: ...,
});
```
The default is 15 minutes if you omit it.

This expiration time is an observable property of the URL as it is included in the query part: &Expires=.... And when it expires it gives back an AccessDenied error:
```
<Error>
	<Code>AccessDenied</Code>
	<Message>Request has expired</Message>
	...
</Error>
```
The second way URLs expire is when the signer loses access to the operation. In the case of roles it is when the credentials expire.
```
const credentials = new AWS.ChainableTemporaryCredentials({
	params: {
		RoleArn: roleArn,
		DurationSeconds: ...,
	}
});
```
It defaults to 1 hour.

When the credentials expire it gives back a different error:
```
<Error>
	<Code>ExpiredToken</Code>
	<Message>The provided token has expired.</Message>
	...
</Error>
```
Since access keys for IAM users do not expire you might not even notice there is a problem during development.

**The two ways of expirations both determine how long the URLs are valid. If the signature is valid for 15 minutes but the session has only 5 minutes remaining the effective duration is only 5 minutes. This is usually a problem for URLs signed close to the end of the session which is one of the worst kind of bugs to debug.**

Fortunately, there is a parameter that controls the role refresh behavior. It is called expiryWindow and it defines how many seconds before expiration the role should be refreshed. In effect, the credentials will be valid for at least this many seconds.
```
const credentials = new AWS.ChainableTemporaryCredentials({
	params: {RoleArn: roleArn}
});
credentials.expiryWindow = 15 * 60; // 15 minutes
```
`const s3 = new AWS.S3({credentials});`
Setting this property makes sure the credentials will expire no sooner than the signature on the URL.

**Comment by noamavner-ma** [comment]
Hey @Omrilu-ma,
The problem with what you sent me is related to AWS V2 SDK.
The AWS V3 SDK is supposed to have a built in Credentials to allow having signed URLs for up to 1 week.

As of right now, we are using the logic to get up to 1 week with `s3RequestPresigner.getSignedUrl` as mention in the AWS documentation, and instead of supporting the signedUrl to be more 1 week, we are getting 24 hours on **all** signedUrl at max.

**Comment by RanVaknin** [comment]
Hi @noamavner-ma ,

Thanks for reaching out. 
From looking at the presigned URL the SDK presigns it with the correct value for `X-Amz-Expires`:

```js
import { S3Client, GetObjectCommand } from "@aws-sdk/client-s3";
import { getSignedUrl } from "@aws-sdk/s3-request-presigner";

async function generatePresignedUrl() {
  const bucketName = "foo-bucket-999";
  const key = "sample.xlsx";

  const client = new S3Client({ region: "us-east-1" }); 
  const command = new GetObjectCommand({ Bucket: bucketName, Key: key });

  try {
    const url = await getSignedUrl(client, command, { expiresIn: 604800 }); 
    console.log(url)
  } catch (error) {
    console.error("Error generating presigned URL", error);
  }
}

generatePresignedUrl();
```

This will output:
```
https://foo-bucket-999.s3.us-east-1.amazonaws.com/sample.xlsx?
X-Amz-Algorithm=AWS4-HMAC-SHA256&
X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&
X-Amz-Credential={{PRINCIPAL_ID}}/20231218/us-east-1/s3/aws4_request&
X-Amz-Date=20231218T231423Z&
X-Amz-Expires=604800&           <--- correct header value
X-Amz-Signature=REDACTED&
X-Amz-SignedHeaders=host&x-id=GetObject
```

Im wondering if the set of credentials you are using to presign this request is of a temporary credentials, or is it signed with an actual user's key and secret.  From the S3 documentation:

> ### Who can create a presigned URL
> 
> Anyone with valid security credentials can create a presigned URL. But for someone to successfully access an object, the presigned URL must be created by someone who has permission to perform the operation that the presigned URL is based upon.
> 
> The following are the types of credentials that you can use to create a presigned URL:
> 
> IAM instance profile – Valid up to 6 hours.
> 
> AWS Security Token Service – Valid up to maximum 36 hours when signed with long-term security credentials or the duration of the temporary credential, whichever ends first.
> 
> IAM user – Valid up to 7 days when you're using AWS Signature Version 4.
> 
> To create a presigned URL that's valid for up to 7 days, first delegate IAM user credentials (the access key and secret key) to the method you're using to create the presigned URL.
> 
> Note
> If you created a presigned URL using a temporary credential, the URL expires when the credential expires. This is true even if the URL was created with a later expiration time. For temporary security credentials lifetimes, see [Comparing AWS STS API operations](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_request.html#stsapi_comparison) in the IAM User Guide.

You can look at the principal ID that you signed the request with by inspecting the URL under the `X-Amz-Credentials` header:
`X-Amz-Credential={{PRINCIPAL_ID}}/20231218/us-east-1/s3/aws4_request&`

It would be helpful if you include the presigned URL provided in v2 and in v3 so we can see if anything is obviously wrong.

Thanks,
Ran~

**Comment by noamavner-ma** [comment]
Hey @RanVaknin the value we are getting in the `X-Amz-Expires` is correct 600000, 1 week (ish)
But we are still getting token expired after 1 day or even less

**Comment by RanVaknin** [comment]
Hi @noamavner-ma,

Please look at the example I provided and the documentation I linked. There is a chance you are creating this presigned URL using temporary credentials which means they will only be valid for validity of those credentials regardless of the `expiresIn` provided value. 

Please inspect the v2 and v3 presigned URLs and compare the principal ID of each one,
Specifically inspect the `X-Amz-Credential={{PRINCIPAL_ID}}/20231218/us-east-1/s3/aws4_request&` part of your presigned URL. This Principal ID(or aws_access_key_id) will tell you what entity the SDK uses to sign the request.

Thanks
Ran~
